fiftyone/core/state.py

```diff
@@ -201,18 +205,59 @@ def get_dataset_stats(dataset):
                 }
             }
     """
-    _sample_doc_cls = type(dataset.name, (foo.DatasetSampleDocument,), {})
+    if isinstance(dataset_or_view, fod.Dataset):
+        view = dataset_or_view.view()
+    else:
+        view = dataset_or_view
+
+    _sample_doc_cls = type(view.dataset_name, (foo.DatasetSampleDocument,), {})
     num_default_fields = len(_sample_doc_cls.get_field_schema())
 
-    field_names = [field_name for field_name in dataset.get_field_schema()]
+    field_schema = view.get_field_schema()
+    custom_fields_schema = OrderedDict(
+        {
+            k: field_schema[k]
+            for k in list(field_schema.keys())[num_default_fields:]
```

Comment:
@tylerganter the way custom fields are retrieved is not deterministic. Requesting that these keys are extracted in a way that all of the current current tests pass.

On the test that is failing, I'm seeing custom fields vary.

Sometimes `metadata` and `filepath` end up as keys here in the python3.5 server tests.

Reply:
thanks for diagnosing this @benjaminpkane . I'll work on a fix

Reply:
interesting...so `field_schema` is an `OrderedDict` and in python 3.5 `list(field_schema.keys())` is not deterministic. I made a solution that does not rely on order